### ISSUE TITLE
Put model in eval() mode in Trainer.predict()

### ISSUE THREAD

**Issue by dsantiago** [PyTorchLightning/pytorch-lightning#7049]
https://github.com/PyTorchLightning/pytorch-lightning/blob/4c07ab5e99dd20c1f309d9e73cdaacc1ebad9499/pytorch_lightning/trainer/trainer.py#L797-L828

I really think this block of code should have, even tough it say in the comments: `#enable eval mode + no grads` it's just disabling the grads but no eval() mode is being called.

```python
model.eval()

{** block_of_test_loop  (the block of code above) **}

model.train()
```

**Comment by awaelchli** [comment]
`model.eval()` happens inside `self.predict_loop.on_predict_model_eval() `

https://github.com/PyTorchLightning/pytorch-lightning/blob/7b0b0d284494d08e3983321d0cc42fe9e5faeb41/pytorch_lightning/core/hooks.py#L183

I'm wondering why you are reporting this. Have you made a test in your LightningModule that your normalization/dropout layers have eval or train mode on?

**Comment by dsantiago** [comment]
> `model.eval()` happens inside `self.predict_loop.on_predict_model_eval() `
> 
> https://github.com/PyTorchLightning/pytorch-lightning/blob/7b0b0d284494d08e3983321d0cc42fe9e5faeb41/pytorch_lightning/core/hooks.py#L183
> 
> I'm wondering why you are reporting this. Have you made a test in your LightningModule that your normalization/dropout layers have eval or train mode on?

Just ignore this one(opened firstly as another contributor told me to do), i already see that it invokes eval on the callbacks... i will close this one, we can talk better on the discussion forum...
